gpflow/monitor/base.py

```diff
@@ -77,30 +77,34 @@ class MonitorTaskGroup:
     `MonitorTask` instances.
     """
 
-    def __init__(self, task_or_tasks: Union[List[MonitorTask], MonitorTask], period: int = 1):
+    def __init__(
+        self, task_or_tasks: Union[Collection[MonitorTask], MonitorTask], period: int = 1
+    ) -> None:
         """
         :param task_or_tasks: a single instance or a list of `MonitorTask` instances.
             Each `MonitorTask` in the list will be run with the given `period`.
         :param period: defines how often to run the tasks; they will execute every `period`th step.
             For large values of `period` the tasks will be less frequently run. Defaults to
             running at every step (`period = 1`).
         """
-        self.tasks = task_or_tasks
+        self._tasks: Collection[MonitorTask] = []
+        self.tasks = task_or_tasks  # type: ignore
         self._period = period
 
     @property
-    def tasks(self) -> List[MonitorTask]:
+    def tasks(self) -> Collection[MonitorTask]:
```

Comment:
Since you're converting it to a list anyway, wouldn't it be more polite to return it as one too (or at least as a `Sequence`)?